clang/test/Driver/sycl-clang-offload-bundler-opaque-pointers.cpp

```diff
@@ -0,0 +1,55 @@
+///
+/// Checks that opaque-pointers are being set for the llvm context accordingly to the bundle targets,
+/// such that for multiple targets, if the first bc input module uses typed pointers, the bitcode
+/// reader will not fail because opaque pointers cannot be converted to typed pointers.
+///
+
+#include <sycl/sycl.hpp>
```

Comment:
This test won't work unless SYCL project is built before. 
There is much simpler way to get a device code - use SYCL_EXTERNAL attribute (see https://github.com/intel/llvm/blob/sycl/clang/test/Driver/sycl-bitfield-accesses.cpp as an example).

Reply:
@GeorgeWeb, please, remove the dependency on DPC++ headers from clang tests.

Reply:
Absolutely. I was on it earlier but I could not find a nice way due to:
```
// RUN: %clangxx -fsycl -target x86_64-unknown-linux-gnu %s -c -o %s.host.o
```
I need `host.o` as an input to `clang-offload-bundler`. I may need to rework the test to not be tied to DPC++/sycl at all as I get `error: unknown type name 'SYCL_EXTERNAL'` when I have to compile and link the object.

Do you have any suggestions? @bader Thanks!

Reply:
How about `__attribute__((sycl_device))`` ?

Reply:
Suggestion from @sarnex should work.

You can also define the macro for the host
```c++
#ifndef __SYCL_DEVICE_ONLY__
#define SYCL_EXTERNAL
#endif
```

On the other hand, it looks like the test is not specific to SYCL at all. You use SYCL mode to create inputs for clang-offload-bundler, but you can use regular C++ mode to create inputs as well. You just need a LLVM bitcode file and an object file. Am I right?
I think we can use the C code from https://github.com/intel/llvm/blob/sycl/clang/test/Driver/clang-offload-bundler.c, but instead of global variable, pass a pointer to the `test_func` to test opaque pointers.